app/views/handlers/list_repeating_question.py

```diff
@@ -2,36 +2,64 @@
 
 from flask import url_for
 
-from app.views.handlers.list_action import ListAction
+from app.questionnaire import Location
+from app.views.handlers.list_edit_question import ListEditQuestion
 
 
-class ListRepeatingQuestion(ListAction):
+class ListRepeatingQuestion(ListEditQuestion):
     @cached_property
     def repeating_block_ids(self) -> list[str]:
-        return self._schema.get_repeating_block_ids()
+        return self._schema.repeating_block_ids
 
-    def get_next_location_url(self) -> str:
+    def get_previous_location_url(self):
+        """
+        return to previous location, or when return to is None, navigate to the previous repeating block
+        unless this is the first repeating block, in which case, route back to the edit block
+        """
         if self._is_returning_to_section_summary():
             return self.get_section_summary_url()
 
-        if first_incomplete_block := self.get_first_incomplete_repeating_block_location_for_list_item(
-            repeating_block_ids=self.repeating_block_ids,
-            section_id=self.current_location.section_id,
-            list_item_id=self.current_location.list_item_id,
-            list_name=self.current_location.list_name,
+        if self.return_to and self.router.can_access_location(
+            Location(
+                section_id=self.current_location.section_id,
+                block_id=self.return_to_block_id,
+            ),
+            routing_path=self._routing_path,
         ):
-            repeating_block_url = url_for(
+            return self._get_location_url(
+                block_id=self._return_to_block_id,
+                anchor=self._return_to_answer_id,
+            )
+
+        repeating_block_index = self.repeating_block_ids.index(
+            self.current_location.block_id
+        )
+        if repeating_block_index != 0:
+            previous_repeating_block_id = self.repeating_block_ids[
+                repeating_block_index - 1
+            ]
+            return url_for(
                 "questionnaire.block",
-                list_name=first_incomplete_block.list_name,
-                list_item_id=first_incomplete_block.list_item_id,
-                block_id=first_incomplete_block.block_id,
+                list_name=self.current_location.list_name,
+                list_item_id=self.current_location.list_item_id,
+                block_id=previous_repeating_block_id,
                 return_to=self._return_to,
                 return_to_answer_id=self._return_to_answer_id,
                 return_to_block_id=self._return_to_block_id,
             )
-            return repeating_block_url
 
-        return super().get_next_location_url()
+        edit_block = self._schema.get_edit_block_for_list_collector(
+            self.parent_block["id"]
+        )
+        return url_for(
```

Comment:
Depending on which merges first, this or Pete's. We need to handle edit block not existing and return to parent if not.

Reply:
@petechd FYI.